hal/inc/socket_hal.h

```diff
@@ -92,7 +92,21 @@ sock_result_t socket_accept(sock_handle_t sd);
  */
 sock_handle_t socket_handle_invalid();
 
+/*
+ * Join a multicast address for all UDP sockets. This will allow reception of multicast packets
+ * sent to the given address on all UDP sockets which have bound the port to which the multicast
+ * packet was sent.
+ * @param addr IP multicast address to join
+ * @return Return the result of the join operation
+ */
+sock_result_t socket_join_multicast(const sockaddr_t *addr);
```

Comment:
Okay, just saw that all the inet_hal.cpp have the reserved parameters and use the HAL_IPAddress. Maybe I should move the functions there, then it would fit better in style. On the other hand it would fit less with the rest of the UDP socket stuff.